src/CarListComponentItem.tsx

```diff
@@ -0,0 +1,78 @@
+import { useState } from "react";
+import { CARS } from "./data/Car"
+
+interface Props{
+    searchCar: string
+    showCars: Function
+}
+
+const CarListItemsComponent = (params: Props) => {
+
+    const deleteCar = (key: number) =>
+    {
+        console.log(key);
+        setCars(cars.filter(x => (x.key === key) === false));
+    }
+    const [cars, setCars] = useState(CARS.filter(data => data.name.toLowerCase().indexOf({params}.params.searchCar.toLowerCase()) > -1).map((data, key) => {
+        return (
+                <CarListItemComponent
+                    image={data.image}
+                    AC={data.AC}
+                    name={data.name}
+                    pricePerDay={data.pricePerDay}
+                    seats={data.seats}
+                    doors={data.doors}
+                    deleteCar={deleteCar}
+                    index={key}                       
+                />)}));
+    return <div>
+            {{cars}.cars.filter(x => x.props.name.toLowerCase().includes({params}.params.searchCar.toLowerCase()))}
+    </div>
+}
+
+interface CarComponentProps
+{
+    name: string;
```

Comment:
it's best to destructure your objects late, so it would be better to provide 'car' as property, not it's fields